respx/router.py

```diff
@@ -99,8 +99,12 @@ def reset(self) -> None:
             route.reset()
 
     def assert_all_called(self) -> None:
-        if any(not route.called for route in self.routes):
-            raise AllCalledAssertionError("RESPX: some routes were not called!")
+        not_called_routes = {str(route) for route in self.routes if not route.called}
+        if not_called_routes:
+            formatted_not_called_routes = "* " + "\n* ".join(not_called_routes)
```

Comment:
Maybe we can make use of regular failed assertion output, instead of manual rendering 🤔 

I'm thinking that this would give us the desired output:

```py
assert not_called_routes == set(), "RESPX: some routes were not called!"
```